File: ruby/37.rb
require "prime"

Prime.lazy.select do |x|
  (0...x.to_s.size).all? {|i| x.to_s[0..i].to_i.prime? && x.to_s[-i..-1].to_i.prime?}
end.drop(4).take(11).to_a.reduce(:+)

File: haskell/37.hs
import Data.Numbers.Primes
import Data.Digits
import Data.List

bothTruncateable x = all (isPrime . (unDigits 10)) list
                 where
                   list = (init $ tails (digits 10 x)) ++ (tail (inits (digits 10 x)))
result = sum $ take 11 $ drop 4 $ filter bothTruncateable primes
